apps/wallet/src/background/Keyring.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Ed25519Keypair } from '@mysten/sui.js';
+import { BehaviorSubject } from 'rxjs';
+import Browser from 'webextension-polyfill';
+
+import { encrypt, decrypt } from '_shared/cryptography/keystore';
+import { generateMnemonic } from '_shared/utils/bip39';
+
+import type { Keypair } from '@mysten/sui.js';
+
+const STORAGE_KEY = 'vault';
+
+class Keyring {
+    private readonly _locked = new BehaviorSubject<boolean>(true);
+    private _encryptedMnemonic: Promise<string | null>;
+    private _keypair: Keypair | null = null;
+
+    constructor() {
+        this._encryptedMnemonic = this.loadMnemonic();
```

Comment:
I thought this was a bit clearer. Load the mnemonic once and handle everything in memory + store when needed. What flows you think it will clean up?

Reply:
I think the creation / import flow would be cleaner because we wouldn't ever need to manage this variable. This is really only used in decrypt (and in wallet initialized, but that could work through just inspecting storage vs having to actually pull values from storage). Given there's really only one place this is read I think it's cleaner to just read it there.

Reply:
> inspecting storage vs having to actually pull values from storage
Is there an api for this?

The main reason to keep it in memory was to avoid accessing the storage every time (assuming only keyring changes the storage).  I'll change it if that is cleaner for you.